src/main/java/seedu/address/model/food/Macronutrient.java

```diff
@@ -0,0 +1,26 @@
+package seedu.address.model.food;
+
+public abstract class Macronutrient {
+    protected int amount;
+    protected int caloricMultiplier;
+    protected int totalCalories;
+
+    /**
+     * Represents macronutrients of 3 types
+     * @param amount
+     * @param caloricMultiplier
+     */
+    public Macronutrient(int amount, int caloricMultiplier) {
+        assert amount >= 0;
+        System.out.println("amount cannot be negative");
```

Comment:
Looks like you're using the assertion incorrectly, it should be
`assert amount >= 0 : "amount cannot be negative";` instead.

Currently it will print the error message regardless of whether the assertion passed or failed.

Also a side note: we should use the java `logger` instead of `System.out.println` apparently.